### ISSUE TITLE
[Next 10][Next/Image] displayed image did not change when changed 'src' prop. (With Incremental Static Regeneration)

### ISSUE THREAD

**Issue by TeXmeijin** [vercel/next.js#18895]
# Bug report

## Describe the bug

When using Incremental Static Regeneration(getStaticProps and getStaticPaths),
And using `<Image>` components,
Then I moved pages clicked by `<Link>` components, then 'src' prop was changed, but actually displayed image was not changed.

❌ **When changed src prop, that displayed image was NOT changed**

```
<Image src={article.imageUrl} width={150} height={150} alt='test'></Image>
```

The solution is specify 'key' props.

⭕ **When changed src prop, that displayed image was changed!**

```
<Image key={article.imageUrl} src={article.imageUrl} width={150} height={150} alt='test ok'></Image>
```

## To Reproduce

Reproduce repo is here:
https://github.com/TeXmeijin/Nextjs10-reproduce-image-component-bug

```
yarn
yarn dev
```

Go to '/article/1' and '/article/2' and move to each other by `<Link>` component's anchor links.

Then first image was not changed, but second image was changed.

Is it Next's expected specification? I don't want set 'key' prop to all `<Image>` components.

For details, see this code:
https://github.com/TeXmeijin/Nextjs10-reproduce-image-component-bug/blob/main/pages/article/%5Bid%5D.js

## Expected behavior

Change displayed image when NOT using 'key' props.

## Screenshots

<img width="218" alt="スクリーンショット 2020-11-06 22 56 26" src="https://user-images.githubusercontent.com/7464929/98374565-3280b600-2084-11eb-923c-e83e3c3c6f8d.png">
<img width="205" alt="スクリーンショット 2020-11-06 22 56 22" src="https://user-images.githubusercontent.com/7464929/98374568-33b1e300-2084-11eb-8ee1-3aec3cc667a6.png">

## System information

- Next.js v10.0.1

**Comment by kripod** [comment]
I just encountered the exact same bug and can verify that the `key` workaround solves the problem for now.

**Comment by Timer** [comment]
This has been fixed on `next@^10.0.2-canary.4`.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
